.github/workflows/ci.yml

```diff
@@ -108,25 +108,34 @@ jobs:
           Cython==0.29.* --install-option="--no-cython-compile"
         echo "$(python3 -m site --user-base)/bin" >> $GITHUB_PATH
 
+
+    # GitHub actions currently does not support modifying an already existing
+    # cache. Hence, we create a new cache for each commit with key
+    # cache-${{ runner.os }}-${{ matrix.cache-key }}-${{ github.sha }}. This
+    # will result in an initial cache miss. However, restore-keys will search
+    # for the most recent cache with prefix
+    # cache-${{ runner.os }}-${{ matrix.cache-key }}-, and if found uses it as
+    # a base for the new cache.
+
```

Comment:
This comment only refers to the ccache caching step and not the others since we don't use `${{ github.sha }}` for dependency caching.

Reply:
I changed the comment to not only describe what this does, but also discuss why we do it, and why we don't do it for the other caches.